### PR TITLE
feat: APP-218 add create post to SellOrdersActionsBar

### PR DESCRIPTION
**Pull request by r41ph**
## Description

https://regennetwork.atlassian.net/browse/APP-218

Added a Create Post button in the Project Page actions bar.
Refactored `ProjectDetails.DataStream` to share the `PostFlow` with `SellOrderActionsBar` in `ProjectDetails`.
Add option for Create Post button to be disabled when a project is a draft in Project Page.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

1. As an admin open any project page and verify that the Create Post button in the ActionsBar and the Data Stream work as expected.
2. Check that the Create Post button is disabled if the project hasn't been published (it's a draft)

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by r41ph** [comment]
@erikalogie @clevinson see testing instructions

**Comment by erikalogie** [comment]
> @erikalogie @clevinson see testing instructions

LGTM!

**Comment by r41ph** [comment]
@erikalogie it's because there's no active account Id, the project is a Draft or it doesn't have assigned a location.

**Comment by r41ph** [comment]
On the project page lower pinned navigation maybe it should be greyed out too? 

It'd be helpful for users if we add some feedback with a tooltip or something when the button is greyed out.

**Comment by r41ph** [comment]
@erikalogie as discussed, now the `Create Post` button in the `Project Page` should be disabled and show a tooltip on hover with the text _"You cannot make posts because either the project is still a draft or it doesn't have assigned a location."_ when the project is not published.

**Comment by erikalogie** [comment]
This looks good! Would you mind putting the same tooltip on the greyed out "create post" button on the /profile/projects page?

File: web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx
Comment (blushi) [inline web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx:97]: right now, `isAdmin` would always be false if connected with wallet connect/keplr mobile web, see how it's declared: https://github.com/regen-network/regen-web/blob/b9220b5ed159a3d667337b36d082cf5ecae43e7f/web-marketplace/src/components/templates/ProjectDetails/ProjectDetails.tsx#L361-L364 (since in this case, there's no `activeAccount`)

That means that instead we show the regular action bar with "buy" button.
I've discussed this with @erikalogie and instead, we should just hide the action bar so we should update `isAdmin` declaration adding a check for `wallet.address && wallet.address === onChainProject?.admin === activeAccount?.addr` and then if `loginDisabled` and `isAdmin` we hide the action bar.

Comment (r41ph) [reply web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx:97]: @blushi I'm not sure the expression you've written would work as expected. You mean something like: `  const isAdmin =
    (!!activeAccount?.addr && onChainProject?.admin === activeAccount?.addr) ||
    (!!activeAccount?.id &&
      offChainProject?.adminAccountId === activeAccount?.id) ||
    (wallet?.address && wallet?.address === onChainProject?.admin) ||
    (wallet?.address && wallet?.address === activeAccount?.addr);`

Comment (r41ph) [reply web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx:97]: @blushi please see whole updated code in PR instead of just my previous message

**Comment by r41ph** [comment]
@erikalogie see updated version with tooltip on the greyed out "create post" button on the /profile/projects page

File: web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx
Comment (blushi) [reply web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx:97]: yes sorry I've updated my comment, this would give

```
const isAdmin =
        (!!activeAccount?.addr && onChainProject?.admin === activeAccount?.addr) ||
        (!!activeAccount?.id &&
          offChainProject?.adminAccountId === activeAccount?.id) ||
        (loginDisabled && wallet?.address && wallet?.address === onChainProject?.admin)
```

After second thought, it's probably safer to add `loginDisabled` check as well in there
because it's possible to have a logged in user `activeAccount` with an address `activeAccount.addr` that is different from `wallet.address` (for instance, if the user logged in with email but a different wallet than the one associated to his account is selected in Keplr) and this particular case, we would not want `isAdmin` to be true for a project that the account in Keplr is an admin of.

Comment (r41ph) [reply web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx:97]: This is now updated accordingly. Please review PR when you get a chance.

**Comment by erikalogie** [comment]
> Details

LGTM! Thanks!

File: web-components/src/components/cards/ProjectCard/ProjectCard.tsx
Comment (blushi) [inline web-components/src/components/cards/ProjectCard/ProjectCard.tsx:136]: it's not necessarily a "create post" button, as you can see in the `onClick` implementation, it could also be a "buy" or "view project" button so I would rename it and also extract it into another component so it doesn't remount this on every render

File: web-components/src/components/cards/ProjectCard/ProjectCard.tsx
Comment (blushi) [inline web-components/src/components/cards/ProjectCard/ProjectCard.tsx:5]: do not import variables from web-marketplace into web-components
web-marketplace depends on web-components and it should not be the other way around
also as I had explained, since we set up translation, it's best to provide any displayed strings as props in web-components components.

File: web-components/src/components/cards/ProjectCard/ProjectCard.tsx
Comment (blushi) [inline web-components/src/components/cards/ProjectCard/ProjectCard.tsx:319]: like I said above, since this button is not always a "create post" button, always displaying the `CREATE_POST_DISABLED_TOOLTIP ` tooltip when it's disabled does not make sense.
Check now what it does on the /projects page for instance:

image

I believe we should rather provide the tooltip text as a prop and if that's defined, we display the tooltip

File: web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx
Comment (blushi) [inline web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx:78]: ditto, could you extract this into a separate component?

**Review by blushi — CHANGES_REQUESTED** [review]

**Comment by r41ph** [comment]
@blushi I believe I have addressed all the points. Please have a look.

File: web-marketplace/src/components/templates/ProjectDetails/ProjectDetails.tsx
Comment (blushi) [inline web-marketplace/src/components/templates/ProjectDetails/ProjectDetails.tsx:463]: from now on, when adding new strings that are meant to be displayed/translated, could you use lingui? this way it's less work for devs that are working on the Terrasos project and if some component already got translated, this won't cause untranslated text.
For instance, declare the variable **in web-marketplace** as `const CREATE_POST_DISABLED_TOOLTIP = msg`some text`;` then use it like this `_(CREATE_POST_DISABLED_TOOLTIP)`

Comment (r41ph) [reply web-marketplace/src/components/templates/ProjectDetails/ProjectDetails.tsx:463]: Sure, I’ll start using Lingui. Thanks for the details!

File: web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.CreatePostButton.tsx
Comment (blushi) [inline web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.CreatePostButton.tsx:21]: best to use `Trans` component in this case, check out https://lingui.dev/tutorials/react-patterns (without `msg`)

**Review by blushi — APPROVED** [review]
tACK
just one small remaining comment

also please use rebase to get `dev` updates :)

File: web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.CreatePostButton.tsx
Comment (r41ph) [reply web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.CreatePostButton.tsx:21]: oh yes, I knew about <Trans>, sorry I didn't use it.
Also sorry for the merge, I clicked the Github Merge button...

Comment (blushi) [reply web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.CreatePostButton.tsx:21]: that's ok, I believe the version with `msg` could work also since the content is just a simple string without sub-components but just trying to follow best practices

if you want to merge from github, make sure to open up the dropdown and select the second option "update with rebase", thanks

image

Comment (r41ph) [reply web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.CreatePostButton.tsx:21]: @blushi I've reviewed this and I did use`{_(CREATE_POST)}` based on what I understand is the recommended way as per https://lingui.dev/tutorials/react-patterns#picking-a-message-based-on-a-variable.

Comment (blushi) [reply web-marketplace/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.CreatePostButton.tsx:21]: ok good to merge after conflicts resolution

**Pull request merged**
